Backend/purrfect-match-backend/src/main/java/com/purrfectmatch/restservice/NegativePageNumberException.java

```diff
@@ -0,0 +1,7 @@
+package com.purrfectmatch.restservice;
+
+public class NegativePageNumberException extends RuntimeException {
+  public NegativePageNumberException(Integer pageNumber) {
+    super("Negative page number was " + pageNumber);
```

Comment:
Instead of catching this and quitting, probably just don't catch and add an advice error handler like you're doing for not found errors.